src/modules/previewpane/MonacoPreviewHandler/FileHandler.cs

```diff
@@ -20,7 +20,7 @@ public static string GetLanguage(string fileExtension)
         {
             try
             {
-                JsonDocument a = JsonDocument.Parse(File.ReadAllText(Settings.AssemblyDirectory + "\\languages.json"));
+                JsonDocument a = JsonDocument.Parse(File.ReadAllText(Settings.AssemblyDirectory + "\\monaco_languages.json"));
                 JsonElement list = a.RootElement.GetProperty("list");
                 for (int i = 0; i < list.GetArrayLength(); i++)
```

Comment:
You can replace it with `for (JsonEkement e in list)`. This prevent us for an index out of bounds crash.

Same for the inner loops.

Reply:
Btw. why not doing something like `if (extensionsList.Contains(ext))`

Reply:
> You can replace it with `for (JsonElement e in list)`. This prevent us for an index out of bounds crash.
> 
> Same for the inner loops.

@htcfreek Do you mean foreach? That's not possible, becuase JsonElement is not enumberable.

Reply:
> > You can replace it with `for (JsonElement e in list)`. This prevent us for an index out of bounds crash.
> > Same for the inner loops.
> 
> @htcfreek Do you mean foreach? That's not possible, becuase JsonElement is not enumberable.

Nevermind you can get the enumerator with enumerateArray()

Reply:
Docs say something that you can get enumeration from it. But we can keep it as is.

But I think we don't need the loop for the extensions. It should be possible to use contains method here. Correct?

Reply:
> Docs say something that you can get enumeration from it. But we can keep it as is.
> 
> But I think we don't need the loop for the extensions. It should be possible to use contains method here. Correct?

I tried it, but it doesn't work. It seems there is no way to use the contains method.